prow/config/inrepoconfig.go

```diff
@@ -94,23 +95,59 @@ func defaultProwYAMLGetter(
 		return nil, fmt.Errorf("failed to merge: %v", err)
 	}
 
-	prowYAMLFilePath := path.Join(repo.Directory(), inRepoConfigFileName)
-	if _, err := os.Stat(prowYAMLFilePath); err != nil {
-		if os.IsNotExist(err) {
-			log.Debugf("File %q does not exist.", inRepoConfigFileName)
-			return &ProwYAML{}, nil
-		}
-		return nil, fmt.Errorf("failed to check if file %q exists: %v", inRepoConfigFileName, err)
-	}
+	prowYAML := &ProwYAML{}
 
-	bytes, err := ioutil.ReadFile(prowYAMLFilePath)
-	if err != nil {
-		return nil, fmt.Errorf("failed to read %q: %v", inRepoConfigFileName, err)
-	}
+	log.Debugf("Attempting to read config files under %q.", inRepoConfigDirName)
+	prowYAMLDirPath := path.Join(repo.Directory(), inRepoConfigDirName)
+	if fileInfo, err := os.Stat(prowYAMLDirPath); !os.IsNotExist(err) && fileInfo.IsDir() {
+		mergeProwYAML := func(a, b *ProwYAML) *ProwYAML {
+			c := &ProwYAML{}
+			c.Presets = append(a.Presets, b.Presets...)
 
-	prowYAML := &ProwYAML{}
-	if err := yaml.Unmarshal(bytes, prowYAML); err != nil {
-		return nil, fmt.Errorf("failed to unmarshal %q: %v", inRepoConfigFileName, err)
+			c.Presubmits = append(a.Presubmits, b.Presubmits...)
+
+			c.Postsubmits = append(a.Postsubmits, b.Postsubmits...)
+
+			return c
+		}
+
+		err := filepath.Walk(prowYAMLDirPath, func(p string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if !info.IsDir() && filepath.Ext(p) == ".yaml" {
+				log.Debugf("Reading YAML file %q", p)
+				bytes, err := ioutil.ReadFile(p)
+				if err != nil {
+					return err
+				}
+				partialProwYAML := &ProwYAML{}
+				if err := yaml.Unmarshal(bytes, partialProwYAML); err != nil {
+					return fmt.Errorf("failed to unmarshal %q: %w", p, err)
+				}
+				prowYAML = mergeProwYAML(prowYAML, partialProwYAML)
+			}
+			return err
+		})
+		if err != nil {
+			return nil, fmt.Errorf("failed to read contents of directory %q: %w", inRepoConfigDirName, err)
+		}
+	} else {
+		log.WithField("file", inRepoConfigFileName).Debug("Attempting to get inreconfigfile")
+		prowYAMLFilePath := path.Join(repo.Directory(), inRepoConfigFileName)
+		if _, err := os.Stat(prowYAMLFilePath); err == nil {
+			bytes, err := ioutil.ReadFile(prowYAMLFilePath)
+			if err != nil {
+				return nil, fmt.Errorf("failed to read %q: %w", inRepoConfigFileName, err)
```

Comment:
Also nit: would it be helpful printing `prowYAMLFilePath ` instead? Same below

Reply:
Indeed, changed everywhere, PTAL :)